server/webapp/WEB-INF/rails.new/webpack/views/preferences/add_filter.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+;(function () { // eslint-disable-line no-extra-semi
+  "use strict";
+
+  const m = require("mithril");
+  const _ = require("lodash");
+
+  function AddFilter(filterModels, pipelineModel) {
+    return {
+      oninit() {
+        filterModels.load();
+      },
+      view() {
+        return m("form", {class: "create-notification-filter", onsubmit: filterModels.save},
+          m(Dropdown, {
+            label:      "Pipeline",
+            name:       "pipeline",
+            options:    pipelineModel.pipelines,
+            onchange:   m.withAttr("value", pipelineModel.currentPipeline)
+          }),
+          m(Dropdown, {label: "Stage", name: "stage", defaultOpt: "[Any Stage]", options: pipelineModel.stages}),
+          m(Dropdown, {label: "Event", name: "event", defaultOpt: "All", options: pipelineModel.events}),
+          m("label", m("input", {type: "checkbox", name: "myCheckin", checked: filterModels.myCommits(), onchange: m.withAttr("checked", filterModels.myCommits)}), m("span", "Only if it contains my check-ins")),
+          m("fieldset",
+            m("input", {type: "submit", value: "Add", class: "primary"}),
+            m("input", {type: "reset", value: "Reset"})
+          )
+        );
+      }
+    };
+  }
+
+  const Dropdown = {
```

Comment:
I could, but the select controls in form_helper.js.msx are far more complex than what is needed here.

The `f.select` would generate something along the lines of:
```html
<div class="columns medium-3 large-3 inline">
  <label>Add a new material of type</label>
  <noscript></noscript>
  <select data-prop-name="selected">
    <option value="git">Git</option>
    ...
  </select>
</div>
```

Whereas this very simple Dropdown (I could rename it) outputs just:
```html
<label>
  <span>Pipeline</span>
  <select name="pipeline"><option value="[Any Pipeline]">[Any Pipeline]</option></select>
</label>
```

The `f.select` component is very versatile, in that it handles tooltips, disabling, etc. While I can omit those attrs because I don't need them, a couple of notable downsides to using the `f.select` component still remain:

  - generates unnecessary wrapper `div.column`, with `label` as a sibling to `select` (in many cases, there's an advantage to wrapping the `select` with `label` -- clicking the text automatically focuses on the input, without needing to specify a `for` attribute, which is missing from `f.select`).
  - does not use standard `name` attribute for the `select` element;
  - not sure what the `<noscript>` entry is for?

I guess in the end, because this form is so basic, I would argue to sticking to the simple `Dropdown` component which generates very simple and standard DOM structures that are easily consumable by `FormData()` if needed. I don't need to bend CSS to work with the wrapping `div.column` elements either; it just works with minimal CSS properties.

What do you think?